edx_proctoring/api.py

```diff
@@ -1386,7 +1394,23 @@ def _get_proctored_exam_view(exam, context, exam_id, user_id, course_id):
         })
 
         if not prerequisite_status['are_prerequisites_satisifed']:
-            # do we have failed prerequisites? That takes priority
+            # do we have any declined prerequisites, if so, then we
+            # will auto-decline this proctored exam
+            if prerequisite_status['declined_prerequisites']:
+                # user hasn't a record of attempt, create one now
+                # so we can mark it as declined
+                create_exam_attempt(exam_id, user_id)
+
+                update_attempt_status(
+                    exam_id,
+                    user_id,
+                    ProctoredExamStudentAttemptStatus.declined,
+                    raise_if_not_found=False
+                )
+                return None
```

Comment:
Since we are retruning None from here, the user will be taken directly to the exam. He will not be shown a screen telling him that a prereq was declined, that's why he's not being shown the proctoring option. Just wanted to make sure that this flow is OK.